pkg/storage/sds/restclient/restclient.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package restclient
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/storage/sds/models"
+)
+
+const (
+	failSendGetRequest = "failure while sending GET request: %w"
+
+	failSendRequestForDocument = "failure while sending request to vault %s to retrieve document %s: %w"
+	failReadResponseBody       = "failure while reading response body: %w"
+)
+
+var logger = log.New("SDS-REST-client")
+
+type marshalFunc func(interface{}) ([]byte, error)
+
+// Client is used to interact with an SDS server.
+type Client struct {
+	sdsServerURL string
+	httpClient   *http.Client
+	marshal      marshalFunc
+}
+
+// Option configures the SDS client.
+type Option func(opts *Client)
+
+// WithTLSConfig option is for definition of secured HTTP transport using a tls.Config instance.
+func WithTLSConfig(tlsConfig *tls.Config) Option {
+	return func(opts *Client) {
+		opts.httpClient.Transport = &http.Transport{TLSClientConfig: tlsConfig}
+	}
+}
+
+// New returns a new instance of an SDS client.
+func New(sdsServerURL string, opts ...Option) *Client {
```

Comment:
Please include a link to the SDS spec.

Reply:
Done! I also included the date and version since the spec page is updated frequently.